### PR TITLE
[Improvement] information_schema.columns support COLUMN KEY

### LINKED ISSUE #10250: [Enhancement] information_schema.columns support update column_key.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, the column_key field in this table is not updated and cannot be used when creating a data dictionary.

### Solution

set column_type for (Unique, Duplicate, Aggrgate), then test mysql protocol compatibility.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by stalary**
# Proposed changes

Issue Number: close https://github.com/apache/doris/issues/10250 https://github.com/apache/doris/issues/11119

## Problem Summary:
information_schema.columns support COLUMN KEY

## Checklist(Required)

1. Type of your changes:
    - [x] Improvement
    - [ ] Fix
    - [ ] Feature-WIP
    - [ ] Feature
    - [ ] Doc
    - [ ] Refator
    - [ ] Others: 
2. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
3. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/schema_scanner/schema_columns_scanner.cpp [hunk 1]
```diff
@@ -343,9 +343,9 @@ Status SchemaColumnsScanner::fill_one_row(Tuple* tuple, MemPool* pool) {
     {
         void* slot = tuple->get_slot(_tuple_desc->slots()[16]->tuple_offset());
         StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
-        str_slot->len = strlen("") + 1;
-        str_slot->ptr = (char*)pool->allocate(str_slot->len);
-        memcpy(str_slot->ptr, "", str_slot->len);
+        str_slot->ptr = (char*)pool->allocate(_desc_result.columns[_column_index].comment.length());
+        str_slot->len = _desc_result.columns[_column_index].columnDesc.columnKey.length();
```
Comment (morningman) [inline be/src/exec/schema_scanner/schema_columns_scanner.cpp:347]: You need to check whether the `columnKey` is set, to make it compatible with old version

**Review by morningman — COMMENTED** [review]
You can add regression test for this case

**Review by morningman — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**